File: zrframe/src/main/java/com/example/zrtool/utilsjava/ColorUtils.java
package com.example.zrtool.utilsjava;

/**
 * @Author qiwangi
 * @Date 2023/8/24
 * @TIME 20:50
 */
public class ColorUtils {

    /**
     * @param colorText
     * @return
     */
    public static int[] convertColorTextToRGB(String colorText) {
        int[] rgbValues = new int[3];

        // 移除颜色文本中的非十六进制字符
        colorText = colorText.replaceAll("[^A-Fa-f0-9]", "");

        // 检查颜色文本的长度
        int colorLength = colorText.length();
        if (colorLength == 3 || colorLength == 6) {
            // 处理三个十六进制字符的情况
            if (colorLength == 3) {
                colorText = colorText.charAt(0) + "" + colorText.charAt(0) +
                        colorText.charAt(1) + "" + colorText.charAt(1) +
                        colorText.charAt(2) + "" + colorText.charAt(2);
            }

            // 解析颜色文本为 RGB 值
            try {
                rgbValues[0] = Integer.parseInt(colorText.substring(0, 2), 16);
                rgbValues[1] = Integer.parseInt(colorText.substring(2, 4), 16);
                rgbValues[2] = Integer.parseInt(colorText.substring(4, 6), 16);
            } catch (NumberFormatException e) {
                e.printStackTrace();
            }
        }

        return rgbValues;
    }
}

File: zrframe/src/main/java/com/example/zrtool/utils/ColorUtils.kt
package com.example.zrtool.utils

/**
 * @Author qiwangi
 * @Date 2023/8/28
 * @TIME 10:37
 */
object ColorUtils {
    /**
     * 不带透明度的随机颜色
     */
     fun getRandomHexColor(): String {
        val color = (0..16777215).random() // 生成一个 0 到 16777215 之间的随机整数
        return String.format("#%06X", color) // 将整数转换为 6 位十六进制字符串
    }
    /**
     * 带透明度的随机颜色
     */
    fun getRandomHexColorWithAlpha(): String {
        val color = (0..16777215).random()
        val alpha = (0..255).random()
        return String.format("#%02X%06X", alpha, color)
    }
}